titan-server/src/organizations/routes.rs

```diff
@@ -19,34 +20,57 @@ pub struct FindOrganizationResponse {
     users: Vec<OrganizationUser>
 }
 
-#[get("/<slug>")]
-pub fn find_organization(
-    slug: &RawStr,
-    unkso_main: UnksoMainForums,
-    unkso_titan: TitanPrimary,
-    app_config: State<config::AppConfig>
-) -> Result<Json<FindOrganizationResponse>, status::NotFound<String>> {
-    let organization_result = organizations::organizations::find_by_slug(
-        slug.as_str(),
-        unkso_titan
+#[derive(FromForm)]
+pub struct FindOrganizationByPathRequest {
+    pub group_type: String,
+    pub path: String
+}
+
+#[get("/path?<form..>")]
```

Comment:
There's no reason to be using a query string here and we're not going to use them here. It's also not semantic if we were to expose an API that arbitrary clients could consume.

We know that the clan is likely to never go away from a Division/Intel organization and not likely that we'll go away from branches either. If we do, we can make a breaking change to the API later.

You can use wildcards and match them in the handler like so:

```suggestion
#[get("/organizations/<type>/<unit>?<extras..>")]
pub fn find_organization_by_path(
    type: String,
    unit: String,
    extras: Form<FindOrganizationByPathRequest>,
    titan_db: TitanPrimary
) -> Result<Json<models::TitanOrganization>, status::NotFound<String>> {
...
}
```

I don't care what they're named, but we need to be more sparing in our use of query strings. We want to focus on high level entities and expose them in a Restful way. I think this is a middle ground between reality and what the future holds.